odfuzz/fuzzer.py

```diff
@@ -1115,20 +1115,35 @@ def add_predecessor(self, predecessor_id):
     def build_string(self):
     #TODO refactor rename build_url_part - this creates the parts after /Entity?$filter... etc ; not entire URL to send to Dispatcher.
         self._query_string = self._accessible_entity.path + '?'
-        for option_name in self._order:
-            if option_name.endswith('filter'):
-                filter_data = deepcopy(self._options[option_name[1:]])
-                option_string = build_filter_string(filter_data)
-            elif option_name.endswith('orderby'):
-                orderby_data = self._options[option_name[1:]]
-                orderby_option = OrderbyOption(orderby_data)
-                option_string = OrderbyOptionBuilder(orderby_option).build()
-            elif option_name.endswith('expand'):
-                option_string = ','.join(self._options[option_name[1:]])
-            else:
-                option_string = self._options[option_name[1:]]
-            self._options_strings[option_name[1:]] = option_string
-            self._query_string += option_name[1:] + '=' + option_string + '&'
+        # for option_name in self._order:
+        #     if option_name.endswith('filter'):
+        #         filter_data = deepcopy(self._options[option_name[1:]])
+        #         option_string = build_filter_string(filter_data)
+        #     elif option_name.endswith('orderby'):
+        #         orderby_data = self._options[option_name[1:]]
+        #         orderby_option = OrderbyOption(orderby_data)
+        #         option_string = OrderbyOptionBuilder(orderby_option).build()
+        #     elif option_name.endswith('expand'):
+        #         option_string = ','.join(self._options[option_name[1:]])
+        #     else:
+        #         option_string = self._options[option_name[1:]]
+        #     self._options_strings[option_name[1:]] = option_string
+        #     self._query_string += option_name[1:] + '=' + option_string + '&'
+        if Config.fuzzer.method == "GET":
```

Comment:
This means that the fuzzer can run EITHER in generating GET requests or DELETE requests ONLY. Am I understanding this right?
If yes, then this is definitely something which is not acceptable.  Fuzzer should be able to create get, delete or get+delete requests in one session. This adds new layer of abstraction above the actual query generation - http method selection.  

I would therefore move the "if method ==GET" out of the current code and create two new methods, e.g. def build_get_url_string(self): and def build_delete_url_string(self): - and call those two methods in a third, http method selector; at the moment it is enough to hardcode it 50/50 randomized - but needs to be in one place to improve further.

One env. variable may be hard to really define user-friendly. It may be worth to add env. variable per HTTP method, as discussed in-face.

e.g. ODFUZZ_HTTP_GET (default True) , ODFUZZ_HTTP_DELETE (default False)



OR....

Did you wanted to create the feature in a way, that DirectBuilder will be inicialized with one method only..?  It may be simpler for FIoriDast usage.. since the networking is done there.. but you modiffied the fuzzer.py as well - and added new env variables, therefore CLI usage as well.. and in this mode, it will not be executed twice (or fourth time with POST and PATCH in the future) so both http methods are covered.

!!Last but not least... I don't see any change relevant to odfuzz dispatcher.. the wrapper around requests module. If I understand this correctly, at the moment, with DELETE method set, it will create normal HTTP GET requests with much shorter urls, not actual HTTP DELETE requests.

Reply:
agreed on removing the environment variables instead of one list, with multiple values in boolean format for each HTTP verb. 
The part related to ODFuzz Dispatcher would be part of another PR after this one.

Reply:
This if is problematic, but to move things forward, let's merge it as it is. Proper solution needs much refactoring in existing codebase, see issue #68 where the details are tracked.